From 5e9b9d11ce1662d0de6dd8b5e79e5f8870f7a397 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Wed, 6 Jan 2021 10:20:53 +0100
Subject: [PATCH] compiler.h: don't define inb/outb and friends on mips

The definition relies on IOPortBase, which is only ever set in
hw/xfree86/os-support/bsd/arm_video.c

This caused build failures on linux/mips with GCC 10, because anything
including compiler.h would get its own definition of IOPortBase.
---
 hw/xfree86/common/compiler.h          | 8 ++------
 hw/xfree86/os-support/bsd/arm_video.c | 3 +++
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/hw/xfree86/common/compiler.h b/hw/xfree86/common/compiler.h
index 2b2008b3f4..eb788d3fd7 100644
--- a/hw/xfree86/common/compiler.h
+++ b/hw/xfree86/common/compiler.h
@@ -518,14 +518,10 @@ xf86WriteMmio32Le(__volatile__ void *base, const unsigned long offset,
     barrier();
 }
 
-#elif defined(__mips__) || (defined(__arm32__) && !defined(__linux__))
-#if defined(__arm32__) || defined(__mips64)
+#elif defined(__arm32__) && !defined(__linux__)
 #define PORT_SIZE long
-#else
-#define PORT_SIZE short
-#endif
 
-_X_EXPORT unsigned int IOPortBase;      /* Memory mapped I/O port area */
+extern _X_EXPORT unsigned int IOPortBase;      /* Memory mapped I/O port area */
 
 static __inline__ void
 outb(unsigned PORT_SIZE port, unsigned char val)
diff --git a/hw/xfree86/os-support/bsd/arm_video.c b/hw/xfree86/os-support/bsd/arm_video.c
index dd1020e33f..180d70543f 100644
--- a/hw/xfree86/os-support/bsd/arm_video.c
+++ b/hw/xfree86/os-support/bsd/arm_video.c
@@ -65,6 +65,7 @@
 #include "xf86Priv.h"
 #include "xf86_OSlib.h"
 #include "xf86OSpriv.h"
+#include "compiler.h"
 
 #if defined(__NetBSD__) && !defined(MAP_FILE)
 #define MAP_FLAGS MAP_SHARED
@@ -162,6 +163,8 @@ xf86DisableIO()
 
 #if defined(USE_ARC_MMAP) || defined(__arm32__)
 
+unsigned int IOPortBase;
+
 Bool
 xf86EnableIO()
 {
-- 
2.29.2

